contracts/Governance.sol

```diff
@@ -482,142 +440,24 @@ contract Governance is
-                        expiry
-                    )
-                )
-            ),
-            v,
-            r,
-            s
-        );
-        require(
-            nonce == _nonces[signer]++,
-            'Governance::delegateBySig: invalid nonce'
-        );
-        _delegate(role, votes, delegatee);
-    }
-
-    function undelegate(
-        bytes32 role,
-        uint256 votes,
-        address delegatee
-    ) external virtual roleExists(role) hasRole(role, delegatee) {
-        _undelegate(role, votes, delegatee);
-    }
-
-    function undelegateBySig(
-        bytes32 role,
-        uint256 votes,
-        address delegatee,
-        uint256 nonce,
-        uint256 expiry,
-        uint8 v,
-        bytes32 r,
-        bytes32 s
-    ) external virtual roleExists(role) hasRole(role, delegatee) {
-        require(
-            block.timestamp <= expiry,
-            'Governance::undelegateBySig: signature expired'
-        );
-        address signer = ECDSA.recover(
-            _hashTypedDataV4(
-                keccak256(
-                    abi.encode(
-                        _UNDELEGATION_TYPEHASH,
-                        role,
-                        delegatee,
-                        nonce,
-                        expiry
-                    )
-                )
-            ),
-            v,
-            r,
-            s
-        );
-        require(
-            nonce == _nonces[signer]++,
-            'Governance::undelegateBySig: invalid nonce'
-        );
-        _undelegate(role, votes, delegatee);
-    }
-
-    function _delegate(
-        bytes32 role,
-        uint256 votes,
-        address delegatee
-    ) internal virtual {
-        delegatees[_msgSender()][role] -= votes;
-        delegations[_msgSender()][role][delegatee] += votes;
-        delegatees[delegatee][role] += votes;
-    }
-
-    function _undelegate(
-        bytes32 role,
-        uint256 votes,
-        address delegatee
-    ) internal virtual {
-        delegatees[delegatee][role] -= votes;
-        delegations[_msgSender()][role][delegatee] -= votes;
-        delegatees[_msgSender()][role] += votes;
-    }
-
-    function getVotes(address voter) public view override returns (uint256) {
-        return token.balanceOf(voter);
+    function getVotes(address account) public view virtual returns (uint256) {
+        return IOpenVoting(votesOracle).getVotes(account);
     }
 
     function getVotes(address account, bytes32 role)
         public
         view
-        override
+        virtual
         returns (uint256)
     {
         if (role == EMPTY_ROLE) {
-            uint256 totalVotes = 0;
-            for (uint256 i = 0; i < rolesList.length; ++i) {
-                totalVotes += delegatees[account][rolesList[i]];
-            }
-            return totalVotes;
+            return getVotes(account);
         }
-        return delegatees[account][role];
+        return IRoleVoting(votesOracle).getVotes(account, role);
```

Comment:
suggestion: Cast the votes oracle in the constructor/setter with a type check before the reference is set.

Yes, this would mean keeping either keeping multiple casted addresses for the same address in this initial case, or create a new interface merging the two.